core/resource/resource.go

```diff
@@ -58,8 +75,9 @@ func NewResource(fullName, kind string, store Store, tnnt tenant.Tenant, meta *M
 		return nil, errors.InvalidArgument(EntityResource, "empty resource spec for "+fullName)
 	}
 
-	if meta == nil {
-		return nil, errors.InvalidArgument(EntityResource, "empty resource metadata for "+fullName)
+	if err := meta.Validate(); err != nil {
```

Comment:
can we check first if meta is not nil before calling Validate method?

Reply:
not needed, it's safe